test/syncenginetestutils.h

```diff
@@ -107,14 +108,31 @@ class DiskFileModifier : public FileModifier
         file.open(QFile::WriteOnly);
         file.write(QByteArray{}.fill(contentChar, size));
     }
-    void appendByte(const QString &relativePath) override {
+    void appendByte(const QString &relativePath, char contentChar) override
+    {
         QFile file{_rootDir.filePath(relativePath)};
         QVERIFY(file.exists());
         file.open(QFile::ReadWrite);
-        QByteArray contents = file.read(1);
+        QByteArray contents;
+        if (contentChar)
+            contents.append(&contentChar, 1);
+        else
+            contents = file.read(1);
         file.seek(file.size());
         file.write(contents);
     }
+    void modifyByte(const QString &relativePath, quint64 offset, char contentChar) override
+    {
+        QFile file{ _rootDir.filePath(relativePath) };
+        QVERIFY(file.exists());
+        file.open(QFile::ReadWrite);
+        QByteArray contents;
+        contents.append(&contentChar, 1);
+        file.seek(offset);
+        file.write(contents);
```

Comment:
(nitpick,  there is an overload of write that takes a char*  i thin, so you do not need to create a QByteArray for it)